Library/ENV/4.3/cc

```diff
@@ -194,6 +198,24 @@ class Cmd
   end
 
   def keep?(path)
+    # The logic in this method will eventually become the default,
+    # but is currently opt-in.
+    return keep_orig?(path) unless ENV["HOMEBREW_EXPERIMENTAL_FILTER_FLAGS_ON_DEPS"]
+
+    # first two paths: reject references to Cellar or opt paths
+    # for unspecified dependencies
+    if path.start_with?(cellar) || path.start_with?(opt)
+      @keg_regex =~ path
+      $2 && @deps.include?($2)
```

Comment:
Might be nice to assign `$2` to a variable first so it's clear what it's referring to.